### PR TITLE
[ci] Implement specs for ParsePackageDiff

### PR DESCRIPTION
**Pull request by ChrisBr**

### CODE REVIEW

File: src/api/spec/mixins/parse_package_diff_spec.rb [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    it { expect(subject['old']).to eq({ "project" => "home:Admin", "package"=>"test", "rev"=>"4", "srcmd5"=>"61c8de91f59df43c9ffd1fa9b4a3f055" }) }
+    it { expect(subject['new']).to eq({ "project" => "home:Admin", "package"=>"test", "rev"=>"5", "srcmd5"=>"ca37dc90f6fd88f63db2ac9f1fc5c41c" }) }
+    it { expect(subject['files'][filename]['state']).to eq('changed') }
+    it { expect(subject['files'][filename]['old']).to eq({ "name" => filename, "md5" => "f00a43bbe6d74b350577e5bce2ea5ff7", "size" => "42" }) }
+    it { expect(subject['files'][filename]['new']).to eq({ "name" => filename, "md5" => "3fd7513ed78f95e2be1bb211369bbea3", "size" => "10" }) }
+    it { expect(subject['files'][filename]['diff']).to eq({ "lines" => "1", "_content" => "# the diff" }) }
+  end
+
+  describe '#sorted_filenames_from_sourcediff' do
+    context 'a other file' do
```
Comment (bgeuken) [inline src/api/spec/mixins/parse_package_diff_spec.rb]: *another

File: src/api/spec/mixins/parse_package_diff_spec.rb [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    context 'a diff file' do
+      let(:filename) { 'apache2.diff' }
+      it_behaves_like  "package with a diff"
+    end
+
+    context 'a patch file' do
+      let(:filename) { 'apache2.patch' }
+      it_behaves_like  "package with a diff"
+    end
```
Comment (bgeuken) [inline src/api/spec/mixins/parse_package_diff_spec.rb]: Does it actually make sense to test the various file suffixes, if this just get's copied over to the xml in the test?

File: src/api/spec/mixins/parse_package_diff_spec.rb (hunk 1 above)
Comment (ChrisBr) [reply src/api/spec/mixins/parse_package_diff_spec.rb]: Dropped!

File: src/api/spec/mixins/parse_package_diff_spec.rb (hunk 2 above)
Comment (ChrisBr) [reply src/api/spec/mixins/parse_package_diff_spec.rb]: Dropped as it is already covered in the "more than one file" case. The important part to test is that the filenames are ordered...

**Pull request merged**